### ISSUE TITLE
OpenGL Support

### ISSUE THREAD

**Issue by reduz** [GPUOpen-Tools/Radeon-GPUProfiler#8]
Would love to optimize [Godot Engine](https://github.com/godotengine/godot) better for AMD but, as we focus on being general purpose and prioritize compatibility, we use OpenGL 3.3.

This tool does not seem to support OpenGL debugging, and the previous one (GPU PerfStudio) is unmaintained and unusable (crashes constantly for me). There is simply no way to optimize for AMD for us, then?

**Comment by callan-amd** [comment]
Have you tried using RenderDoc for debugging? We are Adopting this as our preferred frame debugging tool.

________________________________

**Comment by reduz** [comment]
@callan-amd Ah sorry (I fixed my original post a bit ago, but you replied by mail) RenderDoc is for debugging, I need profiling shaders,  occupancy, etc.

**Comment by procedural** [comment]
@reduz arb_shader_clock is all we got :) AMD tools have a long history of ignoring OpenGL (where's GLSL debugger? It never existed in the first place...), so I wouldn't hold any hopes for this.

**Comment by AlrikG** [comment]
OpenGL Support would be awesome

**Comment by ReaperOfSouls1909** [comment]
we need opengl support for yuzu-emu project as amd performs really bad and nothing else is helping

**Comment by AluminumHaste** [comment]
Would also love to have OpenGL support. I'm a contributor on The Dark Mod, and having better profiling tools.
Unfortunately, AMD has been plagued with poor OpenGL performance for what seems like forever, and they don't seem to care.
Would also love to be able to figure out why new games like Wrath: Aeon of Ruin perform so poorly compared to older Nvidia hardware that's nowhere near as powerful.

**Comment by Degerz** [comment]
I think it's time to close this issue now since there's no future with OpenGL and the industry are working towards deprecating it in the end.

**Comment by Swyter** [comment]
Yeah, you'd be surprised to see the amount of GL 3.0 Core code out there. It's even more important to optimize legacy platforms than modern ones. Having a deeper understanding and better tools is always something to strive for, so if you don't need it, that's fine. ¯\\\_(ツ)_/¯

We want to support as many customers as possible in a cross-platform way and OpenGL is (still) the right tool for the job in Windows, Linux, macOS and Nintendo Switch. By jumping between five to six programs with different pros and cons we can get half of a decent tool.

If anything OpenGL needs better debug layers and reporting.
You're all pretty well covered in Vulkan and DirectX land.

**Comment by Degerz** [comment]
> Yeah, you'd be surprised to see the amount of GL 3.0 Core code out there. It's even more important to optimize legacy platforms than modern ones.

Frankly, many IHVs would vehemently disagree with this sentiment so good luck with trying to convince them of your case ...

> We want to support as many customers as possible in a cross-platform way and OpenGL is (still) the right tool for the job in Windows, Linux, macOS and Nintendo Switch.

That's hardly true anymore. OpenGL applications are constantly getting broken on macOS so there's no guarantee that what an author ships today will work after the next update. The Nintendo Switch is not a good reason to use OpenGL since the platform has a relatively short lifespan which is why many developers would rather opt to use NVN instead. If you wanted to support as 'many' customers in the literal sense then you wouldn't be using OpenGL since it was never designed with industry consensus in mind hence it's shoddy state of support from most vendors.

**Comment by Swyter** [comment]
Okay, that's your subjective view based on theory, while I'm talking about tangible, shipping products.
Because this isn't going anywhere and we have stated our positions, have fun with your lobbying.

**Comment by Degerz** [comment]
If "shipping products" involves supporting legacy code then hardware vendors aren't interested and sure enough we might as well agree to disagree.

**Comment by tuket** [comment]
AMD perfstudio works with OpenGL but it's deprecated and I can't find where to download it for windows.
